Composer/packages/lib/shared/src/copyUtils/copyInputDialog.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { InputDialog } from '../types';
+
+import { ExternalApi } from './ExternalApi';
+import { shallowCopyAdaptiveAction } from './shallowCopyAdaptiveAction';
+
+export const copyInputDialog = async (input: InputDialog, externalApi: ExternalApi): Promise<InputDialog> => {
+  const copy = shallowCopyAdaptiveAction(input, externalApi);
+  const nodeId = copy.$designer ? copy.$designer.id : '';
+
+  if (input.prompt) {
+    copy.prompt = await externalApi.copyLgTemplate(input.prompt, nodeId);
+  }
+
+  if (input.unrecognizedPrompt) {
```

Comment:
Similar things like this "unrecoginzedPrompt". Not sure this is required or not, just want to make sure we leverage the type checking as much as possible. 

Because we control the "constructor", we could be able to enforce more strict type (less optional field). Not for just reducing 1 or 2 line codes, more importantly, have the type enforced at first place, will reduce the possible `state` in our system, means we will reduce the overall complexity of the system.